### PR TITLE
Add jsoncpp as a third-party library

### PR DESCRIPTION
**Pull request by jasondegraw**
Add the amalgamated form of jsoncpp as a third-party library.

### CODE REVIEW

**Comment by Myoldmopar** [comment]
Are we all sure we're set on jsoncpp, @nllong @kbenne @jasondegraw?

**Comment by kbenne** [comment]
Fine by me. I'm ready to merge.

**Comment by kbenne** [comment]
I guess if I'm compelled to think up discussion points I would say putting tests not developed by the third party in "third party" is a little off to me.  Those might go better in non third party directories.  Just a thought....

**Comment by jasondegraw** [comment]
@kbenne That's a good point, I'm fine with moving the tests elsewhere. Is there a better place to put them?

**Comment by kbenne** [comment]
https://github.com/NREL/EnergyPlus/tree/develop/tst

**Comment by Myoldmopar** [comment]
I'd like to motion to merge as-is.  We already have a mixed-bag in third-party with Objexx code.  It's sorta third-party, but frankly it's being developed by a team member.  If anything I'd say let's rename the third-party folder to something else that is a catch for simply _non-core-energyplus-source_.  (Whatever that means...)

**Comment by kbenne** [comment]
@Myoldmopar :-1:

**Comment by kbenne** [comment]
Stuart's tests make sense where they are because he is the original author of the library and the tests.  They go together and they are third party because objexx is a product on its own developed and owned by a different entity.  The json tests in third party imply that the original vender is providing them, which is not the case.  The idea of third_party is that the stuff in there is more or less dropped in as the vendor gave it to us.

**Comment by Myoldmopar** [comment]
ok ok ok ok

```
/tst/third_party/jsoncpp
/tst/third_party/<other tests we write _around_ the third party code>
```

**Comment by Myoldmopar** [comment]
Though I'd be inclined to like the third party source to go into src/third_party/\* to match.....

**Comment by kbenne** [comment]
We are way onto the [bike-shed](http://en.wikipedia.org/wiki/Parkinson's_law_of_triviality) but I think just put the tests in tst/jsoncpp/.  If you prefixed every piece of code you wrote with third_party just because it used some third party stuff, that would be everything you wrote ever.

File: CMakeLists.txt [hunk 1]
```diff
@@ -129,6 +129,7 @@ INCLUDE_DIRECTORIES( ${CMAKE_SOURCE_DIR}/third_party/ObjexxFCL/src/ )
 INCLUDE_DIRECTORIES( ${CMAKE_SOURCE_DIR}/third_party/SQLite/ SYSTEM )
 INCLUDE_DIRECTORIES( "${CMAKE_SOURCE_DIR}/third_party/Expat" "${CMAKE_SOURCE_DIR}/third_party/Expat/lib" SYSTEM)
 INCLUDE_DIRECTORIES( ${CMAKE_SOURCE_DIR}/third_party/CLI/ )
+INCLUDE_DIRECTORIES( ${CMAKE_SOURCE_DIR}/third_party/jsoncpp)
```
Comment (kbenne) [inline CMakeLists.txt:132]: I think if you add "SYSTEM" here you might calm that warning that we otherwise don't have any control over.

**Comment by kbenne** [comment]
Or maybe you have to add some flag or another to disable `enumeral and non-enumeral type in conditional expression [enabled by default]`

I'm pretty sure adding SYSTEM will calm warnings from including warning inducing headers in our code.  Might not do anything for warnings in third party cpp.

File: CMakeLists.txt (hunk 1 above)
Comment (jasondegraw) [inline CMakeLists.txt:132]: Which warning is that?

Comment (kbenne) [inline CMakeLists.txt:132]: This is the closest link I can provide

https://nrel.github.io/EnergyPlusBuildResults/EnergyPlus-96f3217ee7344807b957d62e5d778a0662307b0b-x86_64-Linux-Ubuntu-14.04-gcc-4.8-IntegrationCoverage-Debug.html

**Comment by jasondegraw** [comment]
@kbenne I tried SYSTEM and -Wno-enum-compare, but I'm still getting that warning from GCC... any ideas?

**Comment by kbenne** [comment]
Are you sure your -Wno-enum-compare took hold correctly.  Try make VERBOSE=true to make sure you are getting the flags you expect.

**Comment by jasondegraw** [comment]
Yes, at least I think so. Here's the command line that comes out of CMake:

```
C:\Qt\Tools\mingw491_32\bin\g++.exe   -DXML_STATIC -DOBJEXXFCL_ARRAY_NOALIGN 
-pipe -std=c++11 -pedantic -ffor-scope -Wall -Wextra -Wno-unknown-pragmas 
-Wno-unused-but-set-parameter -Wno-unused-but-set-variable -Wno-maybe-uninitialized 
-Wno-enum-compare @CMakeFiles/jsoncpp.dir/includes_CXX.rsp   -o CMakeFiles\jsoncpp.dir\jsoncpp.cpp.obj 
-c C:\Users\jdegraw\Projects\EnergyPlus\third_party\jsoncpp\jsoncpp.cpp
```

**Comment by kbenne** [comment]
@jasondegraw just hit with a sledge hammer and whack out all the warnings.  https://github.com/NREL/EnergyPlus/blob/develop/third_party/SQLite/CMakeLists.txt#L5

**Comment by Myoldmopar** [comment]
:question: Still warning...weird.  @jasondegraw Can you confirm that `-w` this time is being applied at the command line like last time?

**Comment by jasondegraw** [comment]
@Myoldmopar These are different (non-gcc) warnings. I'll see what I can do.

**Comment by jasondegraw** [comment]
@kbenne So I think all of the warnings are taken care of now. Ready to merge?

**Comment by Myoldmopar** [comment]
Did we really decide jsoncpp was the right library?

**Comment by Myoldmopar** [comment]
@jasondegraw Can this merge?  I can push the green button if so.

**Comment by jasondegraw** [comment]
@Myoldmopar Yes, I think so.

**Pull request merged**